clients/unshaded/src/main/java/tachyon/client/file/options/CreateDirectoryOptions.java

```diff
@@ -0,0 +1,122 @@
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.client.file.options;
+
+import tachyon.Constants;
+import tachyon.annotation.PublicApi;
+import tachyon.client.ClientContext;
+import tachyon.client.UnderStorageType;
+import tachyon.client.WriteType;
+import tachyon.thrift.MkdirTOptions;
+
+@PublicApi
+public final class CreateDirectoryOptions {
+  private boolean mAllowExists;
+  private boolean mRecursive;
+  private UnderStorageType mUnderStorageType;
+
+  /**
+   * @return the default {@link CreateDirectoryOptions}
+   */
+  public static CreateDirectoryOptions defaults() {
+    return new CreateDirectoryOptions();
+  }
+
+  private CreateDirectoryOptions() {
+    mRecursive = false;
+    mAllowExists = false;
+    WriteType defaultWriteType =
+        ClientContext.getConf().getEnum(Constants.USER_FILE_WRITE_TYPE_DEFAULT, WriteType.class);
+    mUnderStorageType = defaultWriteType.getUnderStorageType();
+  }
+
+  /**
+   * @return the under storage type
+   */
+  public UnderStorageType getUnderStorageType() {
+    return mUnderStorageType;
+  }
+
+  /**
+   * @return the allowExists flag value; it specifies whether an exception should be thrown if the
+   *         directory being made already exists
+   */
+  public boolean isAllowExists() {
+    return mAllowExists;
+  }
+
+  /**
+   * @return the recursive flag value; it specifies whether parent directories should be created if
+   *         they do not already exist
+   */
+  public boolean isRecursive() {
+    return mRecursive;
+  }
+
+  /**
+   * @param allowExists the allowExists flag value to use; it specifies whether an exception
+   *        should be thrown if the directory being made already exists.
+   * @return the updated options
+   */
+  public CreateDirectoryOptions setAllowExists(boolean allowExists) {
+    mAllowExists = allowExists;
+    return this;
+  }
+
+  /**
+   * @param recursive the recursive flag value to use; it specifies whether parent directories
+   *        should be created if they do not already exist
+   * @return the updated options
+   */
+  public CreateDirectoryOptions setRecursive(boolean recursive) {
+    mRecursive = recursive;
+    return this;
+  }
+
+  /**
+   * @param writeType the write type to use
+   * @return the updated options
+   */
+  public CreateDirectoryOptions setWriteType(WriteType writeType) {
+    mUnderStorageType = writeType.getUnderStorageType();
+    return this;
+  }
+
+  /**
+   * @return the name : value pairs for all the fields
+   */
+  @Override
+  public String toString() {
+    StringBuilder sb = new StringBuilder("CreateDirectoryOptions(");
```

Comment:
Could we simplify this with https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/builder/ToStringBuilder.html?